stetho/src/main/java/com/facebook/stetho/inspector/protocol/module/DOM.java

```diff
@@ -208,6 +216,103 @@ public void run() {
     });
   }
 
+  @ChromeDevtoolsMethod
+  public PerformSearchResponse performSearch(JsonRpcPeer peer, final JSONObject params) {
+    return mDOMProvider.postAndWait(new UncheckedCallable<PerformSearchResponse>() {
+      @Override
+      public PerformSearchResponse call() {
+        final PerformSearchRequest request = mObjectMapper.convertValue(
+            params,
+            PerformSearchRequest.class);
+
+        // Searches are case insensitive so convert
+        // the query to lower case for simpler comparisons.
+        final String lowerCaseQuery = request.query.toLowerCase();
+
+        final Object root = mShadowDOM.getRootElement();
+
+        // Each search action has a unique ID so that
+        // it can be queried later.
+        final String id = UUID.randomUUID().toString();
+
+        final ArrayList<Integer> results = new ArrayList<>();
+
+        findMatches(lowerCaseQuery, root, results);
+
+        mSearchResults.put(id, results);
```

Comment:
This is a bit awkward to be doing on the main thread.  Strongly prefer logically dividing the work that has to be on the main thread with the rest:

```java
  List<> results = postAndWait(() -> {
    findMatches(...);
    return results;
  });

  mSearchResults.put(id, results);
  PerformSearchResponse response = ...;
  return response;
```java

This is not only a perf issue in general, it also helps clarify your intention for moving to the main thread for some of your work.